addons/website/controllers/main.py

```diff
@@ -703,32 +703,51 @@ def google_maps_api_key(self):
     # Themes
     # ------------------------------------------------------
 
+    # TODO Remove this function in master because it only stays here for
+    # compatibility.
     def _get_customize_views(self, xml_ids):
-        View = request.env["ir.ui.view"].with_context(active_test=False)
-        if not xml_ids:
-            return View
-        domain = [("key", "in", xml_ids)] + request.website.website_domain()
-        return View.search(domain).filter_duplicate()
-
+        self._get_customize_data(self, xml_ids, True)
+
+    def _get_customize_data(self, keys, isViewData):
+        model = 'ir.ui.view' if isViewData else 'ir.asset'
+        Data = request.env[model].with_context(active_test=False)
```

Comment:
`records` here was beter imo, easier to figure than `Data`, which seems to be an `ir.model.data` as we usually write it
(same on other methods, Data is kinda confusing/unusual)

Reply:
It's not a recordset, it's the same convention we use everywhere for naming "model object" (which are "empty recordset"). I guess I would have prefered `Model` to `Data` but I did not want to rewrite everything

Reply:
EDIT: the other `Data` a few line below should have been named `records` though, maybe you wanted to comment that one ^^